### ISSUE TITLE
Improve support for out/ref, lazy evaluation, and overload resolution scenarios

### ISSUE THREAD

**Issue by rjmurillo** [rjmurillo/moq.analyzers#526]
Three improvements:

- Add support for out/ref parameter validation (deferred - requires more complex analysis)
- Add lazy evaluation pattern validation (deferred - complex static analysis needed)
- Validate overload resolution scenarios (deferred - requires semantic analysis improvements)

See #514 for more details

**Comment by rjmurillo** [comment]
## Recommendation: split into 3 concrete issues; re-scope item 1

Ground-truthing the three bundled items against current `main` (not training data):

**Item 1 (out/ref parameter validation) is already partly shipped for callbacks.** `CallbackSignatureShouldMatchMockedMethodAnalyzer` (Moq1100) compares parameter ref-kind exactly:

```csharp
// src/Analyzers/CallbackSignatureShouldMatchMockedMethodAnalyzer.cs:228
return mockedParameter.RefKind == parameterSymbol.RefKind;
```

and the fixer emits `ref`/`out`/`in` modifiers (`CallbackSignatureShouldMatchMockedMethodFixer.cs:198-204`). So `out`/`ref`/`in` mismatches in `Callback` signatures are already diagnosed. What remains unspecified is out/ref validation *outside* callbacks (e.g. `Setup`/`Returns` argument matchers with `out`/`ref`), which is a narrower, definable scope.

**Items 2 and 3 have no concrete failure cases, examples, or diagnostic IDs.** Building them from the current description would be a speculative implementation, which this repo's quality bar explicitly disallows ("No speculative implementations"). Each needs: a bad-code sample, the expected diagnostic message, and a proposed Moq-rule ID before any analyzer work starts.

### Proposed disposition
This is a backlog-management call for the owner, so I'm recommending, not closing:

1. **Split** this into three tracked issues, each with a concrete failure case + proposed rule ID:
   - `out`/`ref` matcher validation **outside** callbacks (item 1 remainder; note Moq1100 already covers callbacks).
   - Lazy-evaluation pattern validation (`Returns(() => v)` vs `Returns(v)`) — define the intended diagnostic.
   - Overload-resolution validation in `Setup` — define the ambiguity case.
2. Keep this issue as the umbrella/epic, or close it once the three are filed.

No code change is merited until each split item carries a concrete, testable failure case.
